### PR TITLE
Stanage MPI builds: document compile + runtime tweaks via EasyBuild hooks

### PR DESCRIPTION
**Pull request by willfurnass**
I like the idea of this info being public.  Can make it public via other means if/when we publish our EB `hooks.py` file (inc explanatory comments).

### CODE REVIEW

File: stanagepilot/software/parallel/openmpi.rst [hunk 1]
```diff
@@ -139,7 +139,26 @@ Your output would be something like: ::
...
+     (via an environment variable - ``SLURM_MPI_TYPE=pmix_v4``) that 
+     PMIx version 4 is to be used for launching remote processes using ``srun``.
+   * Versions greater than 4.0.0 are configured at runtime to use 
+     LibFabric (OFI) for inter-process communications, which in turn is 
+     configured at runtime via environment variables to use the PSM2 OFI provider 
+     for efficient OmniPath support:
+     * ``OMPI_MCA_pml=cm``
+     * ``OMPI_MCA_mtl=ofi``
+     * ``OMPI_MCA_mtl_ofi_provider_include=psm2``
+     * ``PMIX_MCA_psec=native``
```
Comment (jkwmoore) [inline stanagepilot/software/parallel/openmpi.rst:161]: We might just want to include why we're doing that (CentOS 7, older PSM2 lib.)

Otherwise LGTM.

Comment (jkwmoore) [reply stanagepilot/software/parallel/openmpi.rst:161]:
```
        # With OpenMPI >=4.1.0 the PSM2 lib provided by the CentOS 7 repos is
        # too old so just use LibFabric (OFI), which the EB EasyConfigs/
        # EasyBlock enable by default with newer OpenMPIs.
 ```

File: stanagepilot/software/parallel/impi.rst [hunk 2]
```diff
@@ -137,3 +137,16 @@ This will generate output of the form:
...
     Hello world: rank 1 of 2 running on node051.pri.stanage.alces.network
+
+Installation notes
+------------------
+
+This section is primarily for administrators of the system. Intel MPI has been installed using the default Easybuild config files but with the following tweaks made via EasyBuild hooks:
+
+* Modulefiles are patched so that
+  * they instruct Slurm at runtime 
+    (via ``SLURM_MPI_TYPE=pmi2``) that
```
Comment (jkwmoore) [inline stanagepilot/software/parallel/impi.rst:148]: Looks like this indentation here is causing issues, suspect this needs to be level with the bullet point?

File: stanagepilot/software/parallel/openmpi.rst (hunk 1 above)
Comment (willfurnass) [reply stanagepilot/software/parallel/openmpi.rst]: Ta; addressed that.

**Comment by jkwmoore** [comment]
I'll have a quick gander with VS code now.

**Comment by jkwmoore** [comment]
Last commit should fix things up as shown below:

image


image

**Review by jkwmoore — APPROVED** [review]
LGTM. Please whack merge on there if you're happy with the edits I made.

**Pull request merged**